dashboard/app/views/home/index.html.haml

```diff
@@ -29,17 +29,18 @@
   - if current_user
     #user_hero
       = render template: 'api/user_hero'
-    - current_user.unadvertised_user_scripts.each_slice(2) do |course_pair|
-      .row
-        - course_pair.each do |user_script|
-          = render partial: 'shared/course_progress_block', locals: { script: user_script.script, user_script: user_script }
+    - if Gatekeeper.allows('logged_in_studio_homepage', default: true)
+      - current_user.unadvertised_user_scripts.each_slice(2) do |course_pair|
+        .row
+          - course_pair.each do |user_script|
+            = render partial: 'shared/course_progress_block', locals: { script: user_script.script, user_script: user_script }
 
   - else # no logged in user
     = render partial: 'home/sign_in_buttons'
 
   = render partial: 'home/all_courses'
 
-  - if current_user && !current_user.teacher?
+  - if current_user && !current_user.teacher? && Gatekeeper.allows('logged_in_studio_homepage', default: true)
```

Comment:
Is it slightly better to put the Gatekeeper check first here? (would `current_user.teacher?` ever cause a query?)